error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java

```diff
@@ -1204,12 +1204,15 @@ StepVerifier.FirstStep<? extends T> after(Flux<T> flux) {
     }
   }
 
-  /** Don't unnecessarily call {@link StepVerifier.Step#expectNext(Object[])}. */
-  static final class StepVerifierStepExpectNextEmpty<T> {
+  /**
+   * Don't unnecessarily call {@link StepVerifier.Step#expectNext(Object[])} or {@code
+   * StepVerifier.Step#expectNextCount(0L)}.
+   */
+  static final class StepVerifierStepIdentity<T> {
     @BeforeTemplate
     @SuppressWarnings("unchecked")
     StepVerifier.Step<T> before(StepVerifier.Step<T> step) {
-      return step.expectNext();
+      return Refaster.anyOf(step.expectNext(), step.expectNextCount(0L));
```

Comment:
Technically, empty `Iterable`s in a call to `StepVerifier.Step#expectNextSequence(Iterable<T>)` would also qualify, see suggestion.

```suggestion
      return Refaster.anyOf(
          step.expectNext(), step.expectNextCount(0L), step.expectNextSequence(ImmutableSet.of()));
```

However, IIUC we'd need to have a different `before` step which matches empty collections. Let's not apply this as part of this PR and keep it bite-sized. :wink:

Reply:
Nice one. This relies on an `IsEmpty` `Matcher` I've been meaning to finalize (see also a comment elsewhere in this class). Will add a note :)